include/bmat8.hpp

```diff
@@ -193,6 +193,7 @@ class BMat8 {
     size_t nr_rows() const;
     //! Returns a \c std::vector for rows of \c this
     std::vector<uint8_t> rows() const;
+    void rows(std::array<uint8_t, 8>&) const;
```

Comment:
Just returning a `std::array<uint8_t, 8>` instead of a `std::vector<uint8_t>` should remove all the allocation that motivate the change of API.

Reply:
Right, but I don't think it's possible to overload functions in C++ by return type, so simply adding a member function with return type `std::array<uint8_t, 8>` won't work. For backwards compatibility I suggest not removing the member function `std::vector<uint8_t> rows() const`, maybe the new member function could be called `rows_array` or something similar?

Reply:
I didn't payed attention to the fact that you where keeping the old function. `rows_array` looks good to me. Then `rows` should be an alias for a `rows_vector`.